internal/databases/postgres/simple_tar_ball_composer.go

```diff
@@ -0,0 +1,100 @@
+package postgres
```

Comment:
I don't see any need for the additional `TarBallComposer` here. `SimpleTarBall` composer does not introduce any new way to pack tarballs, just alters the `BundleFiles` behavior.

So I suggest removing the new composer and passing the SimpleBundleFiles to the regular composer. 

Just replace the `*RegularBundleFiles` with the `BundleFiles` interface [here](https://github.com/wal-g/wal-g/blob/3a5120f1cc208a69db6a834c979a4c97a4d2e7bc/internal/databases/postgres/regular_tar_ball_composer.go#L18) so the regular tarball composer could support both `SimpleBundleFiles` and `RegularBundleFiles`.

Reply:
As for `TarFileSets`, you can do the similar:
1. Create the `TarFileSets` interface
2. Create `SimpleTarFileSets`/`NopTarFileSets` (new behavior) and `RegularTarFileSets` (old behavior)